crates/sui-gateway/src/read_api.rs

```diff
@@ -1,122 +1,31 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::{collections::BTreeMap, path::Path, sync::Arc, time::Duration};
-use sui_storage::IndexStore;
-use sui_types::{crypto::get_key_pair, sui_serde::Base64};
-
 use crate::{
     api::{RpcGatewayServer, TransactionBytes},
     rpc_gateway::responses::{ObjectResponse, SuiTypeTag},
 };
 use anyhow::anyhow;
 use async_trait::async_trait;
 use jsonrpsee::core::RpcResult;
-use sui_config::{NetworkConfig, PersistedConfig};
+use std::sync::Arc;
+use sui_core::gateway_state::GatewayTxSeqNumber;
 use sui_core::{
-    authority::{AuthorityState, AuthorityStore},
-    authority_active::{gossip::gossip_process, ActiveAuthority},
+    authority::AuthorityState,
     gateway_types::{
         GetObjectInfoResponse, SuiObjectRef, TransactionEffectsResponse, TransactionResponse,
     },
-    sui_json::SuiJsonValue,
 };
-use sui_core::{authority_client::NetworkAuthorityClient, gateway_state::GatewayTxSeqNumber};
+use sui_json::SuiJsonValue;
 use sui_types::base_types::{ObjectID, SuiAddress, TransactionDigest};
-use tracing::info;
+use sui_types::sui_serde::Base64;
 
-pub struct SuiFullNode {
+pub struct ReadApi {
```

Comment:
`ReadApi` is perhaps a bit of a generic name, even if the contents of this struct - a pointer to the authority store- are. I would add a code comment here indicating the first expected customer is expected to be the full node. This should also help folks grepping for 'full node'.